Add support for html5 boolean attributes to content_tag and tag.

ijcd opened the pull request:
Fixes https://github.com/phoenixframework/phoenix_html/issues/182

I also updated the helper tags so that the current output of using `checked="checked"` does not change. Direct use of these tags would change, however. I also updated tests and doctests to reflect this.

HTML5 boolean attributes can come in several forms.

These mean checked is true:
```
     <checkbox checked="checked">
     <checkbox checked="">
     <checkbox checked>
```

This means checked is false:

```
     <checkbox>
```

josevalim commented on lib/phoenix_html/tag.ex:
```diff
@@ -89,8 +91,11 @@ defmodule Phoenix.HTML.Tag do
+    for a <- attrs do
+      case a do
+        {k, v} -> [?\s, k, ?=, ?", attr_escape(v), ?"]
+        {k} -> [?\s, k]
```
We should not wrap in a tuple with one element. Just pass the value as is.

josevalim commented on lib/phoenix_html/form.ex:
```diff
@@ -738,6 +738,11 @@ defmodule Phoenix.HTML.Form do
 
+    opts =
+      Keyword.update(opts, :checked, false, fn c ->
+        boolean_attribute("checked", c)
```
Question: Why do we need this? Can't we just pass the boolean forward?

ijcd replied:
Passing the boolean forward will mean the naked attribute version will be output instead. I added the boolean_attribute() helper to retain the current behavior for `form_for`, `select`, `radio_button`, etc. They currently emit the boolean form `checked="checked"` and I thought it would be less impactful for that not to change. That is still a valid boolean form. If we just pass the boolean forward, then many of the current tests need to be updated, and people might be surprised that the html generation changed. Either works -- this is just less disruptive, I feel.

josevalim replied:
I see. Let's go ahead and rip the band-aid, otherwise people may be confused why some attributes work like that and others do not. :)

ijcd commented:
I removed the `boolean_attribute` helper and updated the tests. Would you normally squash out the changes from PR discussion or merge as is? I haven't contributed to this repo yet.

josevalim commented:
We can squash from the github UI now, so no worries! :D

josevalim commented:
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:
